instrumentation/action_mailer/lib/opentelemetry/instrumentation/action_mailer/instrumentation.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActionMailer
+      # The Instrumentation class contains logic to detect and install the ActionMailer instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('6.1.0')
+        install do |_config|
+          resolve_email_address
+          require_dependencies
+        end
+
+        present do
+          defined?(::ActionMailer)
+        end
+
+        compatible do
+          gem_version >= MINIMUM_VERSION
+        end
+
+        option :disallowed_notification_payload_keys, default: [],  validate: :array
+        option :notification_payload_transform,       default: nil, validate: :callable
+        option :email_address,                        default: :omit, validate: %I[omit include]
+
+        private
+
+        def gem_version
+          ::ActionMailer.version
+        end
+
+        def resolve_email_address
+          return unless ActionMailer::Instrumentation.instance.config[:email_address] == :omit
+
+          ActionMailer::Instrumentation.instance.config[:disallowed_notification_payload_keys] += %i[to from bcc cc]
+
+          if ActionMailer::Instrumentation.instance.config[:notification_payload_transform].nil?
+            mask_email = lambda do |payload|
+              payload[:mail].gsub!(/(From:\s+|To:\s+|Cc:\s+)[^\s]+/, '\1[redacted]')
+              payload
+            end
+          else
+            original_callable = ActionMailer::Instrumentation.instance.config[:notification_payload_transform]
+            mask_email = lambda do |payload|
+              original_callable.call(payload)
+              payload[:mail].gsub!(/(From:\s+|To:\s+|Cc:\s+)[^\s]+/, '\1[redacted]')
```

Comment:
Is this the full contents of the email message? 

If it is, we should not include this at all. It would be the equivalent of recording the contents of an HTTP response body.

Reply:
Yes, the problem is that when omit the email address, the attributes one can be excluded, but the content still have the email address

```
"Date: Fri, 23 Feb 2024 15:34:28 +0000\r\n" +
    "From: no-reply@example.com\r\n" +
    "To: test_mailer@otel.org\r\n" +
    "Message-ID: <65d8bb04c68d8_15bcc45779@4d4fcd81426e.mail>\r\n" +
    "Subject: Welcome to OpenTelemetry!\r\n" +
    "Mime-Version: 1.0\r\n" +
...
```
With gsub:
```
"Date: Fri, 23 Feb 2024 15:34:28 +0000\r\n" +
    "From: [redacted]\r\n" +
    "To: [redacted]\r\n" +
    "Message-ID: <65d8bb04c68d8_15bcc45779@4d4fcd81426e.mail>\r\n" +
    "Subject: Welcome to OpenTelemetry!\r\n" +
    "Mime-Version: 1.0\r\n" +
```

Reply:
> Is this the full contents of the email message?

> If it is, we should not include this at all. It would be the equivalent of recording the contents of an HTTP response body.

Will remove the mail part.